m4/with_libz.m4

```diff
@@ -0,0 +1,21 @@
+AC_DEFUN([PBS_AC_WITH_LIBZ],
+[
+  AC_ARG_WITH([libz],
+    AS_HELP_STRING([--with-libz=DIR],
+      [Specify the directory where libz is installed.]
+    )
+)
+AS_IF([test "x$with_libz" != "x"],
```

Comment:
Please add a comment here to explain what these macros are doing

Reply:
I have added a comment.